### PR TITLE
sql: fix RSG test problems

### PR DESCRIPTION
**Pull request by madelynnblue**
- support remaining function paramater types
- prevent transactions from starting and failing

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9373)

### CODE REVIEW

File: sql/rsg_test.go [hunk 1]
```diff
@@ -108,8 +115,8 @@ func TestRandomSyntaxFunctions(t *testing.T) {
 				var v interface{}
 				switch typ.(type) {
 				case *parser.DInt:
-					i := r.Intn(math.MaxInt64)
-					i -= r.Intn(math.MaxInt64)
+					i := int64(r.Intn(math.MaxInt64))
+					i -= int64(r.Intn(math.MaxInt64))
```
Comment (knz) [inline sql/rsg_test.go]: Don't make it so complicated :)
just do `r.Int63()`.

File: sql/rsg_test.go [hunk 2]
```diff
@@ -120,6 +127,24 @@ func TestRandomSyntaxFunctions(t *testing.T) {
...
 					v = fmt.Sprintf(`'%s'`, t.Format(time.RFC3339Nano))
+				case *parser.DBool:
+					if r.Intn(2) == 0 {
+						v = "false"
+					} else {
+						v = "true"
+					}
+				case *parser.DDate:
+					i := int64(r.Intn(math.MaxInt64))
+					i -= int64(r.Intn(math.MaxInt64))
```
Comment (knz) [inline sql/rsg_test.go]: Ditto

**Review by knz — CHANGES_REQUESTED** [review]
Don't you also want to exclude syntax using `SAVEPOINT`?

**Comment by mjibson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/rsg_test.go, line 119 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9373#-KRe0ft1EA_XsRJR3-2o-r1-119:-KRi2M94bIDLUIc55UZ8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/78d456772e7240c7b2df1047f8d9254ab52df8d9/sql/rsg_test.go#L119)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Don't make it so complicated :)
> 
> just do `r.Int63()`.
> 
> 
</details>
Done.

---

*[sql/rsg_test.go, line 138 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9373#-KRe0ft1EA_XsRJR3-2o-r1-138:-KRi2MtuzHePB_QXmdE-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/78d456772e7240c7b2df1047f8d9254ab52df8d9/sql/rsg_test.go#L138)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Ditto
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9373)*

**Review by knz — APPROVED** [review]

**Comment by knz** [comment]
LGTM

**Comment by mjibson** [comment]
The tests passed without ignoring SAVEPOINT, so I think it's ok.

**Pull request merged**